### PR TITLE
[WIP] Publish documentation for Quickwit CLI

### PR DESCRIPTION
**Pull request by guilload**
Work in progress PR for the Quickwit CLI documentation. (#24)

TODO:
- [x] getting started
- [x] `new` command
- [x] `index` command
- [x] `search` command
- [x] `serve` command
- [x] `delete` command
- [ ] CI pipeline that automatically publishes new changes to `docs.quickwit.io`

### CODE REVIEW

File: docs/quickwit-cli/README.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+# Quickwit CLI
+
+Quickwit CLI is a tool that makes it easy to index and search structured or unstructured data from the command line. It consumes datasets consisting of newline-delimited JSON objects with arbitrary keys and produces indexes that can be stored locally or remotely on an object storage such as Amazon S3 and queried with subsecond latency.
+
+Key features include the following:
+- schemaless
+- local and remote indexes
+- native support for time-series datasets
```
Comment (fulmicoton) [inline docs/quickwit-cli/README.md:8]: I think people imagine something different when they hear time-series datasets. Maybe "native support for timestamps" (though I agree it sounds dumb)

File: docs/quickwit-cli/README.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+- no support for file formats other than JSON
+- no support for object storages not compatible with Amazon S3
+- no faceted search
+- no deletions (append mode only)
+
+Influence our roadmap by voting in our [GitHub issues](https://github.com/quickwit-inc/quickwit/issues) for the features that you need the most!
+
+### Warning
+
+Cloud providers charge for data transfers in and out of their networks. In addition, querying an index from a remote machine adds some extra latency. For those reasons, we recommend that you test and use the Quickwit CLI from an instance located within your cloud provider's network.
```
Comment (fulmicoton) [inline docs/quickwit-cli/README.md:20]: and offers a very bad throughput.

File: docs/quickwit-cli/README.md (hunk 1 above)
Comment (fmassot) [reply docs/quickwit-cli/README.md:8]: It's more related to time partitionning or sharding
maybe put "native support for time sharding"

File: docs/quickwit-cli/README.md (hunk 2 above)
Comment (fmassot) [reply docs/quickwit-cli/README.md:20]: on the limitation, one is important to tell: schema canont evolved unless you reindex everything.

**Comment by fmassot** [comment]
PR merge. We need to define a way to udpate the quickwit.io though.

**Pull request merged**